pkg/cloud/services/networking/securitygroups_rules.go

```diff
@@ -283,16 +288,45 @@ func GetSGWorkerAllowAll(remoteGroupIDSelf, secControlPlaneGroupID string) []inf
 	}
 }
 
-func GetSGControlPlaneGeneral(remoteGroupIDSelf, secWorkerGroupID string) []infrav1.SecurityGroupRule {
+func getSGControlPlaneCNI(openStackCluster *infrav1.OpenStackCluster, remoteGroupIDSelf, secWorkerGroupID string) []infrav1.SecurityGroupRule {
+	cni := openStackCluster.Spec.SecurityGroupForCNI
+	if len(cni) == 0 {
+		cni = DEFAULTCNI
+	}
+
+	switch cni {
+	case CALICOCNI:
+		return getSGControlPlaneCalico(remoteGroupIDSelf, secWorkerGroupID)
+	default:
+		return []infrav1.SecurityGroupRule{}
```

Comment:
The CRD description describes that a warning is printed when the value is unknown.
I think this warning could be added here.